### ISSUE TITLE
sway crashes when opening netrw in neovim

### ISSUE THREAD

**Issue by jtheoof** [swaywm/wlroots#1811]
I am not really sure how you help you guys to pinpoint the issue, but I reproduce this very easily on my machine. Considering the stack trace below, it looks like the issue lies within `wlroots`. The issue does not occur with `vim`. Only with `neovim`.

Distribution: Arch Linux
Sway: 1.2-2
Terminal: alacritty

Launch `neovim`. Type `:Explore` to launch `netrw-:Explore` Wait a few seconds, sway crashes with the following stack trace.

```
Sep 01 21:47:07 ansan audit[5401]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=10 pid=5401 comm="sway" exe="/usr/bin/sway" sig=11 res=1
Sep 01 21:47:07 ansan kernel: sway[5401]: segfault at 8 ip 00007f53be8ce1b0 sp 00007ffde40e4268 error 4 in libwayland-server.so.0.1.0[7f53be8c8000+7000]
Sep 01 21:47:07 ansan kernel: Code: 0e 48 83 c3 08 49 39 dd 77 d0 b8 01 00 00 00 48 83 c4 08 5b 5d 41 5c 41 5d c3 48 89 3f 48 89 7f 08 c3 0f 1f 84 00 00 00 00 00 <48> 8b 47 08 48 89 3e 48 89 46 08 48 89 77 08 48 8b 46 08 48 89 30
Sep 01 21:47:07 ansan kernel: audit: type=1701 audit(1567388827.643:336): auid=1000 uid=1000 gid=1000 ses=10 pid=5401 comm="sway" exe="/usr/bin/sway" sig=11 res=1
Sep 01 21:47:07 ansan systemd[1]: Started Process Core Dump (PID 6912/UID 0).
Sep 01 21:47:07 ansan kernel: audit: type=1130 audit(1567388827.653:337): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@41-6912-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Sep 01 21:47:07 ansan audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@41-6912-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Sep 01 21:47:07 ansan xdg-desktop-por[6540]: Error reading events from display: Broken pipe
Sep 01 21:47:07 ansan systemd[1262]: xdg-desktop-portal-gtk.service: Main process exited, code=exited, status=1/FAILURE
Sep 01 21:47:07 ansan systemd[1262]: xdg-desktop-portal-gtk.service: Failed with result 'exit-code'.
Sep 01 21:47:07 ansan at-spi-bus-launcher[16976]: X connection to :0 broken (explicit kill or server shutdown).
Sep 01 21:47:08 ansan systemd-coredump[6913]: Process 5401 (sway) of user 1000 dumped core.
                                              
                                              Stack trace of thread 5401:
                                              #0  0x00007f53be8ce1b0 wl_list_insert (libwayland-server.so.0)
                                              #1  0x00007f53be85fe7f wlr_seat_keyboard_enter (libwlroots.so.3)
                                              #2  0x000055a68b053368 n/a (sway)
                                              #3  0x000055a68b046a47 n/a (sway)
                                              #4  0x000055a68b0652b7 n/a (sway)
                                              #5  0x000055a68b049477 n/a (sway)
                                              #6  0x000055a68b05226a n/a (sway)
                                              #7  0x00007f53be886cfe n/a (libwlroots.so.3)
                                              #8  0x00007f53be87526d wlr_keyboard_notify_key (libwlroots.so.3)
                                              #9  0x00007f53be850197 n/a (libwlroots.so.3)
                                              #10 0x00007f53be84f10c n/a (libwlroots.so.3)
                                              #11 0x00007f53be8cb7f2 wl_event_loop_dispatch (libwayland-server.so.0)
                                              #12 0x00007f53be8ca39c wl_display_run (libwayland-server.so.0)
                                              #13 0x000055a68b028592 n/a (sway)
                                              #14 0x00007f53beec9ee3 __libc_start_main (libc.so.6)
                                              #15 0x000055a68b0288be n/a (sway)
                                              
                                              Stack trace of thread 5403:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5405:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5409:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5408:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5404:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5407:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5412:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5406:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5410:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5411:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5417:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5416:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5414:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5413:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5415:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
                                              
                                              Stack trace of thread 5418:
                                              #0  0x00007f53be78f415 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f53bc8e69bc n/a (radeonsi_dri.so)
                                              #2  0x00007f53bc8e65b8 n/a (radeonsi_dri.so)
                                              #3  0x00007f53be78957f start_thread (libpthread.so.0)
                                              #4  0x00007f53bef9f0e3 __clone (libc.so.6)
```

**Comment by RPigott** [comment]
On arch I can reproduce with packages sway 1.2-2 + wlroots 0.7.0-1
I _cannot_ reproduce in sway-git r6047.aa8ea2fa-1 (git-describe 1.2-rc1-43-gaa8ea2fa) + wlroots-git 0.7.0.r2.gbf90474b-1 (git-describe 0.7.0-2-gbf90474b)

**Comment by RPigott** [comment]
[debug log](https://github.com/swaywm/wlroots/files/3586034/sway-1.2.log)
[backtrace w/ debug](https://github.com/swaywm/wlroots/files/3586035/sway-1.2.txt)

**Comment by RedSoxFan** [comment]
This is/was a sway issue. Duplicate of swaywm/sway#4526. Fixed by swaywm/sway#4525

**Comment by jtheoof** [comment]
I tried the patch you mentioned @RedSoxFan. It fixed the issue. Thanks!
